### DISCUSSION TITLE
Would wp-api-privacy trigger hammering protection?

### DISCUSSION THREAD

**Discussion by drzimp402** [wp-privacy/wp-api-privacy#12] [General]
A thought came to mind today.  Could this plugin potentially trigger hammering protection from wp.org?  Now, I don't know if .org has any hammering protection installed/enabled, but here's the hypothetical that came to mind.

Imagine a single web host that has a ton of WP installs. Then a large number of these individual installs add the wp-api-privacy plugin, changing the user agent to 'WordPress/Private'. From there, when these sites hit wp.org, the requests all come from the same IP address (the web host's) with the same user agent. This could be seen as potential hammering.

Note: For the core/version-check request, the requests could also have the same payload sent to wp.org. 

So I'm wondering if instead of "WordPress/Private", should it be something a bit more unique to each installation, just to prevent potential hammering? Maybe a hash of some sort?

**Comment by duanestorey** [comment]
I mean, I don't know what they do on their side.  But the user-agent field is usually not even looked at by most applications. In your scenario, the actual number of requests is the same, which isn't anything obscene. WordPress itself only calls home a few times a day I believe, so multiply that by how ever many hosts, and it's not a lot. 

I could add a hash I guess at some point that would be unique.  I've thought about how to do that using the salts in the wp-config.php. But at that point now I'm sending something they don't really understand, so I'm not sure if that's worse than just omitting it. I really am trying hard to be ethical with how I'm doing this. If I just omit data, then can discard it on their side, and it's a win/win (they still get useful data). If I start fudging the data that's being sent, now I'm sort of polluting that data, and I don't really want to do that.  The hash is a good idea, but depending on how they interpret it on their side it could end up in some domain table as well.  Not horrible, but something I've been trying to be aware of.  I've seen other solutions to this privacy problem where they are sending other URLs instead of stripping it - then the data itself starts to become polluted.  

 So I guess I'd prefer to wait for this to actually be a problem, if that makes sense. But it's a good thing to think about, thanks.
